lib/AlexaSkill.js

```diff
@@ -114,11 +115,15 @@ class AlexaSkill {
         }
       }
 
-
       if (this.getOnLaunchRequestHandlers().length === 0) {
         throw new Error('onLaunchRequest must be implemented');
       }
 
+      if (alexaEvent.request.type === 'CanFulfillIntentRequest') {
```

Comment:
Why the need to handle this as an intent?

Reply:
Every intent can be fulfilled. To verify that, every intent with their slots should have the logic if they need to be fulfilled or not. That's why I convert this request to an `IntentRequest` and add this boolean flag so developers know if they're going to handle this canFulfillRequest. If they deny, the flow ends and the state machine sends the response back to the Alexa service. If they do accept to fulfill it, the state machine starts over the process with this flag set to `undefined`

Reply:
I'm not sure that I love that approach. For example, not all intents are going to be on the root level. Also, isn't it odd the co-mingle the logic for how to kick off an intent, and report that it is actionable in one handler?

It seems like this is putting too much work into intent handlers, and with special flags to indicate something is slightly different. Also, there's likely to be only a few intents that you'd want to signal as can fulfill, but the rest should be no by default. This way all of the intents are going to need to do this special check.

Finally, it seems like we should offer a shorthand expression for intents that are fulfillable regardless of slot, then a way to say, let me run a function to be sure.